adapters/flipp/flipp.go

```diff
@@ -0,0 +1,225 @@
+	}
+
+	publisherUrl, err := url.Parse(request.Site.Page)
+	if err != nil {
+		return nil, fmt.Errorf("Unable to parse site url. %v", err)
+	}
+
+	var contentCode string
+	if publisherUrl != nil {
+		contentCode = publisherUrl.Query().Get("flipp-content-code")
+	}
+
+	placement := Placement{
+		DivName: InlineDivName,
+		SiteID:  &flippExtParams.SiteID,
+		AdTypes: getAdTypes(flippExtParams.CreativeType),
+		ZoneIds: flippExtParams.ZoneIds,
+		Count:   &Count,
+		Prebid:  buildPrebidRequest(flippExtParams, request, imp),
+		Properties: &Properties{
+			ContentCode: &contentCode,
+		},
+		Options: flippExtParams.Options,
+	}
+
+	var userKey string
+	if request.User != nil && request.User.ID != "" {
+		userKey = request.User.ID
+	} else if flippExtParams.UserKey != "" {
+		userKey = flippExtParams.UserKey
+	} else {
+		uid, err := uuid.NewV4()
+		if err != nil {
+			return nil, fmt.Errorf("Unable to generate user uuid. %v", err)
+		}
+		userKey = uid.String()
+	}
+
+	keywordsArray := strings.Split(request.Site.Keywords, ",")
+	var keywordsSlice []Keyword
+
+	for _, k := range keywordsArray {
+		keywordsSlice = append(keywordsSlice, Keyword(k))
+	}
+
+	campaignRequestBody := CampaignRequestBody{
+		Placements: []*Placement{&placement},
+		URL:        request.Site.Page,
+		Keywords:   keywordsSlice,
+		IP:         request.Device.IP,
+		User: &CampaignRequestBodyUser{
+			Key: &userKey,
+		},
+	}
+
+	adapterReq, err := a.makeRequest(request, campaignRequestBody)
+	if err != nil {
+		return nil, fmt.Errorf("Make request faile with err %v", err)
+	}
+
+	return adapterReq, nil
+}
+
+func buildPrebidRequest(flippExtParams openrtb_ext.ImpExtFlipp, request *openrtb2.BidRequest, imp openrtb2.Imp) *PrebidRequest {
+	var height int64
+	var width int64
+	if imp.Banner != nil && len(imp.Banner.Format) > 0 {
+		height = imp.Banner.Format[0].H
+		width = imp.Banner.Format[0].W
+	}
+	prebidRequest := PrebidRequest{
+		CreativeType:            &flippExtParams.CreativeType,
+		PublisherNameIdentifier: &flippExtParams.PublisherNameIdentifier,
+		RequestID:               &imp.ID,
+		Height:                  &height,
+		Width:                   &width,
+	}
+	return &prebidRequest
+}
+
+func (a *adapter) MakeBids(request *openrtb2.BidRequest, requestData *adapters.RequestData, responseData *adapters.ResponseData) (*adapters.BidderResponse, []error) {
+	if responseData.StatusCode == http.StatusNoContent {
+		return nil, nil
+	}
+
+	if responseData.StatusCode == http.StatusBadRequest {
+		err := &errortypes.BadInput{
+			Message: "Unexpected status code: 400. Bad request from publisher. Run with request.debug = 1 for more info.",
+		}
+		return nil, []error{err}
+	}
+
+	if responseData.StatusCode != http.StatusOK {
+		err := &errortypes.BadServerResponse{
+			Message: fmt.Sprintf("Unexpected status code: %d. Run with request.debug = 1 for more info.", responseData.StatusCode),
+		}
+		return nil, []error{err}
+	}
```

Comment:
As mentioned above, you can make use of existing util functions for these checks :- 

```
if err := httputil.CheckResponseStatusCodeForErrors(responseData); err != nil {
		return nil, []error{err}
	}
```